src/common/AdaptiveCencSampleDecrypter.h

```diff
@@ -0,0 +1,68 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "../utils/log.h"
+
+#include <bento4/Ap4.h>
+
+class Adaptive_CencSampleDecrypter : public AP4_CencSampleDecrypter
+{
+public:
+  Adaptive_CencSampleDecrypter(AP4_CencSingleSampleDecrypter* single_sample_decrypter,
+                               AP4_CencSampleInfoTable* sample_info_table)
+    : AP4_CencSampleDecrypter(single_sample_decrypter, sample_info_table)
+  {
+  }
+
+  virtual AP4_Result DecryptSampleData(AP4_UI32 poolid,
```

Comment:
please move implementation in to relative .cpp file
also `#include "../utils/log.h"` must be moved to cpp